core/src/actors/sessions_manager.rs

```diff
@@ -0,0 +1,264 @@
+                        act.process_get_peer_response(res)
+                    })
+                    //// Process the socket address received
+                    // This returns a FutureResult containing a success or error
+                    .and_then(|address, _act, _ctx| {
+                        debug!("Trying to create a new outbound connection to {}", address);
+
+                        // Get connections manager from registry and send an OutboundTcpConnect message to it
+                        let connections_manager_addr =
+                            System::current().registry().get::<ConnectionsManager>();
+                        connections_manager_addr.do_send(OutboundTcpConnect { address });
+
+                        actix::fut::ok(())
+                    })
+                    .wait(ctx);
+            }
+
+            // Reschedule the bootstrap peers task
+            act.bootstrap_peers(ctx);
+        });
+    }
+
+    /// Method to process peers manager GetPeer response
+    fn process_get_peer_response(
+        &mut self,
+        response: Result<PeersSocketAddrResult, MailboxError>,
+    ) -> FutureResult<SocketAddr, (), Self> {
+        response
+            // Unwrap the Result<PeersSocketAddrResult, MailboxError>
+            .unwrap_or_else(|_| {
+                debug!("Unsuccessful communication with peers manager");
+                Ok(None)
+            })
+            // Unwrap the PeersSocketAddrResult
+            .unwrap_or_else(|_| {
+                debug!("An error happened in peers manager when getting a peer");
+                None
+            })
+            // Check if PeersSocketAddrResult returned `None`
+            .or_else(|| {
+                debug!("No peer obtained from peers manager");
+                None
+            })
+            // Filter the result checking if outbound address is eligible as new peer
+            .filter(|address: &SocketAddr| {
+                self.sessions.is_outbound_address_eligible(address.clone())
+            })
+            // Check if there is a peer after filter
+            .or_else(|| {
+                debug!("No eligible peer obtained from peers manager");
+                None
+            })
+            // Convert Some(SocketAddr) or None to FutureResult<SocketAddr, (), Self>
+            .map(actix::fut::ok)
+            .unwrap_or_else(|| actix::fut::err(()))
+    }
+}
+
+/// Make actor from `SessionsManager`
+impl Actor for SessionsManager {
+    /// Every actor has to provide execution `Context` in which it can run
+    type Context = Context<Self>;
+
+    /// Method to be executed when the actor is started
+    fn started(&mut self, ctx: &mut Self::Context) {
+        debug!("Sessions Manager actor has been started!");
+
+        // TODO: Call Config Manager --> and set server address and limits
+        self.sessions
+            .set_server_address("127.0.0.1:50000".parse().unwrap());
```

Comment:
consider creating a static/const sever address string?

Reply:
You are also right. We left this hard-coded on purpose because we will wait until we have the `Config Manager` to change hard-coded addresses. :)